genie/genie-controller.go

```diff
@@ -324,13 +325,14 @@ func ParsePodAnnotationsForMultiIPPrefs(client *kubernetes.Clientset, k8sArgs ut
 }
 
 //  parseCNIAnnotations parses pod yaml defintion for "cni" annotations.
-func parseCNIAnnotations(annot map[string]string, client *kubernetes.Clientset, k8sArgs utils.K8sArgs) ([]string, error) {
+func parseCNIAnnotations(annot map[string]string, client *kubernetes.Clientset, k8sArgs utils.K8sArgs, conf utils.NetConf) ([]string, error) {
 	var finalAnnots []string
 
-	if len(annot) == 0 {
-		fmt.Fprintf(os.Stderr, "CNI Genie no annotations is given! Default plugin is weave! annot is %V\n", annot)
-		finalAnnots = []string{"weave"}
-	} else if strings.TrimSpace(annot["cni"]) == "" {
+	if len(annot) == 0 || strings.TrimSpace(annot["cni"]) == "" {
+		if conf.Default != "" {
```

Comment:
I created a pod with NO annotations. 
However conf.Default is: 

Jul 28 18:00:58 karun-cni-dev kubelet[2156]: CNI Genie conf.Default= []

So, it goes to line 336 and it is treated as if annotations with cni: "" is passed. Therefore, the pod never goes to 'Running'. 
@erikstmartin can you verify this please?

Reply:
@erikstmartin , I was syncing with @kshafiee about the current status, I can help u for any support needed. Contact u through hangout

Reply:
@erikstmartin , Have checked the code modifications and also executed some test scenario with these changes, have done slight modifications. It would be nice if we can together discuss on that and can move this towards closure. Will contact u